### DISCUSSION TITLE
[QUESTION] Create related entity with single mutation

### DISCUSSION THREAD

**Discussion by rbiggers** [doug-martin/nestjs-query#1193] [General]
I want to first start off by saying that I am very impressed with what you all have done here with this project! It has made my life so much easier when developing against GraphQL and tight deadlines! So thank you!

I have tried to do my due diligence but can't seem to come up with an answer on this one. I'm sure I am missing something simple but I need your help with a solution. 

I have a simple use case where I have a User entity and in this User entity it has a one-to-one relation with a Profile entity. 

So the question... is there a way that I can create a User entity and have the Profile entity with the relation set from the createOneUser mutation?

Setup:
- NestJs
- TypeOrm
- Nestjs-Query
- CockroachDB

Simple Example:
``` typescript
@InputType('UserInput')
export class UserInput {
  @IsString()
  @MaxLength(20)
  @Field()
  public nickName!: string;

  @Field(() => ProfileInput, {
    nullable: false,
    description: 'The user profile',
  })
  public profile!: ProfileInput;
}
```
``` typescript
@InputType('ProfileInput')
export class ProfileInput {
  @IsString()
  @MaxLength(20)
  @Field({
    nullable: false,
    description: 'The gender of the profile',
  })
  public gender!: string;
}
```
``` typescript
@Entity()
export class UserEntity {
  @Column()
  public nickName!: string;

  @OneToOne(() => ProfileEntity)
  @JoinColumn()
  public profile!: ProfileEntity;
}
```
``` typescript
@Entity()
export class ProfileEntity {
  @Column()
  public gender!: string;
}
```
What I want is a way to create the Profile at the same time the User is created. 

``` json
  "createOneUserInput": {
    "user":{
      "nickName": "Grasshopper",
      "profile": {
        "gender": "MALE"
      }
    }
  }
```
``` gql
mutation createOneUser($createOneUserInput: CreateOneUserInput!) {
  createOneUser(input: $createOneUserInput) {
    id
    nickName
    profile {
      id
      gender
    }
  }
}
```

The API accepts the input above but does not set the relation. Is there something I am missing or need to add (maybe a custom service to manually set the relation)? 

Again I would really appreciate your help on this. 

Thank you for your hard work and time!

**Comment by smolinari** [comment]
Moved to Discussions.

I'm not a TypeORM user (and I'm assuming you're using that), but there is the cascade feature. That might help. 

https://typeorm.io/#/undefined/using-cascades-to-automatically-save-related-objects

Scott

**Comment by rbiggers** [comment]
@smolinari yup I knew I was missing something simple... Thank you for your help!
